### PR TITLE
[fix]: [PLG-2320]: added ResponseTimeRecorder + improved logging + UTs for signup domainDenyList logic

### PR DESCRIPTION
**Pull request by kapil-harness**
## Describe your changes
added ResponseTimeRecorder + improved logging + UTs for signup domainDenyList logic

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/47976)

### CODE REVIEW

**Review by thomas-grasman — DISMISSED** [review]
> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/47976)* status: 2 file(s) to review. 1 of 2 approvals obtained.

File: 920-ng-signup/src/test/java/io/harness/signup/validator/SignupValidatorTest.java [hunk 1]
```diff
@@ -227,4 +231,21 @@ public void testInvalidIntent() throws IOException {
...
+  public void testValidateSignupDomain_withWrongGCSCredentials() {
+    String email = "test@google.com";
+    when(signupDomainDenylistConfiguration.getProjectId()).thenReturn("projectId");
+    when(signupDomainDenylistConfiguration.getBucketName()).thenReturn("bucketId");
+    when(signupDomainDenylistConfiguration.getGcsCreds()).thenReturn("gcsCreds");
+
+    try {
+      signupValidator.validateSignupDomain(email);
+    } catch (Exception ex) {
+      fail("Unexpected exception was thrown while fetching domainDenylist from GCS");
```
[the remark this answers is not in the document]
Comment (neelam-harness) [reply 920-ng-signup/src/test/java/io/harness/signup/validator/SignupValidatorTest.java:247]: Good catch. Yes, using this suggested way is the right way for exception test cases.

**Review by mohitg0795 — DISMISSED** [review]
> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/47976)* status: 1. unresolved discussions.

**Review by kapil-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/47976)* status: 2 file(s) to review. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`920-ng-signup/src/test/java/io/harness/signup/validator/SignupValidatorTest.java` line 247 at r1](https://reviewable.io/reviews/harness/harness-core/47976#gh-1196570952:-NVeoKx_CWrSeFZbNwjw:b4k3e0j) ([raw file](https://github.com/harness/harness-core/blob/f5753c98a47943524be045622c2d800a6444e0b4/920-ng-signup/src/test/java/io/harness/signup/validator/SignupValidatorTest.java#L247)):*
<details><summary><i>Previously, neelam-harness wrote…</i></summary><blockquote>

Good catch. Yes, using this suggested way is the right way for exception test cases.
</blockquote></details>

I have changed this UT, tried to improve the readability.

**Comment by kapil-harness** [comment]
trigger its

**Comment by kapil-harness** [comment]
trigger uts

**Review by thomas-grasman — APPROVED** [review]
> Reviewed 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/47976)* status: 2 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by Xin-Harness — APPROVED** [review]

**Comment by kapil-harness** [comment]
trigger review

**Pull request merged**